src/OrchardCore.Modules/OrchardCore.Autoroute/Models/AutoroutePart.cs

```diff
@@ -20,6 +20,11 @@ public class AutoroutePart : ContentPart
         /// </summary>
         public bool Disabled { get; set; }
 
+        /// <summary>
+        /// Whether the related content item was removed.
+        /// </summary>
+        public bool ContentItemRemoved { get; set; }
```

Comment:
This property is just to track it inside the index provider? 

And then not persisted?

Would it be better to have a scoped service to track removed, rather than adding a property?

Or too much...

Reply:
Yes

> And then not persisted?

I was thinking about retrieving an item version without this state, e.g. with audit trail

> Would it be better to have a scoped service to track removed, rather than adding a property?

> Or too much...

Yes, here we just need a scoped state, i also thought about using our ShellScope items / features
But yes, maybe too much to have a service
i will think about it this night